langchain4j-agentic/src/main/java/dev/langchain4j/agentic/cognisphere/CognisphereRegistry.java

```diff
@@ -0,0 +1,158 @@
+package dev.langchain4j.agentic.cognisphere;
+
+import dev.langchain4j.Internal;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import java.util.ServiceLoader;
+import java.util.Set;
+
+/**
+ * Singleton registry for managing Cognisphere instances.
+ * Provides methods to register, retrieve, and manage Cognisphere objects.
+ * Supports persistence through a pluggable persistence provider.
+ */
+@Internal
+public enum CognisphereRegistry {
```

Comment:
Why is this an enum? Is this how we implement the singleton pattern in langchain4j?

Reply:
That's only the safest and most concise way to implement a singleton in Java. Please let me know if and how I need to change this, but as you said without CDI I don't see any better alternative.

Reply:
No longer a singleton.